### PR TITLE
harmful var for guns

### LINKED ISSUE #36224: Pacifists can't use tasers, but can use stun batons
Seems inconsistent. To me, it'd be more fun if pacifists had access to all non-damaging weaponry. Similar to a non-lethal playthrough.

### PR DESCRIPTION
**Pull request by PKPenguin321**
🆑
fix: Pacifists can now use non-harmful guns, such as tasers, staffs of healing, toy foam dart guns, or medibeams
🆑

support for pacifism to allow firing of non-harmful guns

currently tasers, a few toy foam dart guns, a couple wizard magic wands, and a handful of others (door creation, teleportation, resurrection, floral somatoray, medibeam) are considered non-harmful

Fixes #36224

### CODE REVIEW

**Comment by arsserpentarium** [comment]
need more non harmul guns tbh

**Comment by cebutris** [comment]
Pacifists can hurt people via pen darts with this!

Feature?

**Comment by PKPenguin321** [comment]
Who cares

File: code/modules/projectiles/guns/magic/staff.dm [hunk 1]
```diff
@@ -69,6 +71,7 @@
 	item_state = "honker"
 	max_charges = 4
 	recharge_rate = 8
+	harmful = FALSE
```
Comment (ShizCalev) [inline code/modules/projectiles/guns/magic/staff.dm]: bs

**Comment by cacogen** [comment]
I'm wondering if maybe it'd be better to check the projectile the gun is currently firing rather than the gun itself. It'd lower the amount of edits needed and allow pacifists to use weapons like energy guns when they're set to disable.

The only problem would be things like the staff of chaos where the user shouldn't have foreknowledge of what projectile it's going to fire. It'd need its own variable to prevent pacifists from firing it due to the risk of harm.

File: code/modules/projectiles/guns/magic/staff.dm (hunk 1 above)
Comment (PKPenguin321) [reply code/modules/projectiles/guns/magic/staff.dm]: Afaik all it does is "slip" you which does no harm (its just a stun)

**Comment by XDTM** [comment]
You can check the chambered projectile and it'll work with the staff of chaos. You can kinda meta the next shot like this but it's not the end of the world.

**Comment by CPTANT** [comment]
Its ironic that the taser is probably the most lethal weapon in the game.

**Pull request merged**